c7n/resources/rds.py

```diff
@@ -2093,16 +2097,15 @@ def handle_optiongroup_cache(self, client, paginator, option_groups):
                 if og_values is not None:
                     ogcache[og] = og_values
                     continue
-                option_list = list(itertools.chain(*[p['OptionGroupsList']
+                option_groups_list = list(itertools.chain(*[p['OptionGroupsList']
                     for p in paginator.paginate(OptionGroupName=og)]))
 
                 ogcache[og] = {}
-                for option in option_list:
-                    if option['Options']:
-                        for p in option['Options']:
-                            ogcache[og].update(p)
-                cache.save(cache_key, ogcache[og])
+                for option_group in option_groups_list:
+                    ogcache[og] = option_group
```

Comment:
This is the bug with the old code where we were only storing the last option of the option group. With the new code, we're storing all the options of the option group.